apps/cowswap-frontend/src/modules/permit/hooks/usePreGeneratedPermitInfoForToken.ts

```diff
@@ -0,0 +1,25 @@
+import { Currency } from '@uniswap/sdk-core'
+
+import { Nullish } from 'types'
+
+import { usePreGeneratedPermitInfo } from './usePreGeneratedPermitInfo'
+
+import { IsTokenPermittableResult } from '../types'
+
+/**
+ * Fetch pre-generated permit info for a single token
+ */
+export function usePreGeneratedPermitInfoForToken(token: Nullish<Currency>): {
+  permitInfo: IsTokenPermittableResult
+  isLoading: boolean
+} {
+  const { allPermitInfo, isLoading } = usePreGeneratedPermitInfo()
+
+  const permitInfo =
+    allPermitInfo && token && 'address' in token ? allPermitInfo[token.address.toLowerCase()] : undefined
```

Comment:
ubernit, my mind didn't compute the predicate at first, maybe we can save ' token && 'address' in token' into a const

pls, feel free to ignore, is not imp at all